gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineReleaseCompatibility.java

```diff
@@ -42,13 +43,22 @@
     public void apply(Project project) {
         this.project = project;
 
-        project.getTasks().withType(JavaCompile.class).configureEach(BaselineReleaseCompatibility::configureTask);
+        project.getTasks().withType(JavaCompile.class).configureEach(javaCompile -> {
+            javaCompile.getOptions().getCompilerArgumentProviders().add(new ReleaseFlagProvider(javaCompile));
+        });
     }
 
-    private static void configureTask(JavaCompile javaCompile) {
-        // using a lazy argument provider is crucial because otherwise we'd try to read sourceCompat / targetCompat
-        // before the user has even set it in their build.gradle!
-        javaCompile.getOptions().getCompilerArgumentProviders().add(() -> {
+    // using a lazy argument provider is crucial because otherwise we'd try to read sourceCompat / targetCompat
+    // before the user has even set it in their build.gradle!
+    private static final class ReleaseFlagProvider implements CommandLineArgumentProvider {
```

Comment:
can this be an anonymous class for the sake of readability?

Reply:
I actually want this to be named so that if it ever appears in a stacktrace or warning message we'll know exactly where to look.  The `$0` message before was pretty hard to spot.